app/main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+class Animal:
+    alive = list()
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        if health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        object_items_str = ", ".join(
+            (f"{k.capitalize()}: {v}" for k, v in vars(self).items())
+        )
+        return f"{{{object_items_str}}}"
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, victim: Herbivore) -> None:
+        if (
+                victim.health > 0
+                and isinstance(victim, Herbivore)
```

Comment:
It would be a better approach to check first that the `victim` is an instance of the `Herbivore` class and after that conduct additional checks that rely on its attributes to prevent potential errors in case other objects would be passed to this method.